heat-production-optimization/Pages/DataDisplay.cshtml.cs

```diff
@@ -9,7 +9,7 @@ namespace heat_production_optimization.Pages
     public class DataDisplayModel : PageModel
     {
         private readonly SourceDataDbContext _context = new SourceDataDbContext();
-        public DbSet<HeatDemandDataModel> heatDemandData;
+        public DbSet<HeatDemandDataModel>? heatDemandData;
```

Comment:
I don't know if you had any issues with this but I'd strongly lean on leaving this property non-nullable since we don't want potential null displays